R/distance.R

```diff
@@ -62,37 +68,59 @@
 #'   return(as.dist(res))
 #' }
 #'
-#' seqDist(immdata$data[1:2], .method = f) # the result based on our custom distance defined above
+#' seqDist(immdata$data[1:2], .method = f, .group_by_seqLength = FALSE)
 #' @export seqDist
 
-seqDist <- function(.data, .col = "CDR3.nt", .method = "hamming", ...) {
+seqDist <- function(.data, .col = "CDR3.nt", .method = "hamming", .group_by = c("V.name", "J.name"), .group_by_seqLength = TRUE, ...) {
   .validate_repertoires_data(.data)
   sample_truth <- .data[[1]]
+  gb_absent <- all(is.na(.group_by))
   # Since seqDist works with any columns of string type, classic .col values are not suported
   if (.col %in% c("aa", "nt", "v", "j", "aa+v")) stop("Please, provide full column name")
+  if (!all(.group_by %in% colnames(sample_truth)) && !gb_absent) {
+    stop("Expected column(s): ", paste0(.group_by, collapse = ", "), "; some of them are missing in data!")
+  }
```

Comment:
The message probably should be different in case when grouping variables are not provided at all